x-pack/platform/packages/shared/onechat/onechat-common/tools/definition.ts

```diff
@@ -43,9 +43,12 @@ export interface ToolDefinition<TConfig extends object = Record<string, unknown>
    * The description for this tool, which will be exposed to the LLM.
    */
   description: string;
+  /**
+   * Indicate whether this tool is editable by users or not.
+   */
+  readonly: boolean;
```

Comment:
Tools all expose that `readonly` flag now, which can also be used for the UI to figure out which tool is editable and which is not.

Technically, all user tools are `readonly: false`, all our tools are `readonly: true` (at least for now)